protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/AuthCmdHandler.java

```diff
@@ -465,9 +508,19 @@ private Response doUnknownAuth(SMTPSession session, String authType, String init
     @Override
     public List<String> getImplementedEsmtpFeatures(SMTPSession session) {
         if (session.isAuthAnnounced()) {
+            ImmutableList.Builder<String> authTypesBuilder = ImmutableList.builder();
             if (session.getConfiguration().isPlainAuthEnabled()) {
-                return ESMTP_FEATURES;
+                authTypesBuilder.add(AUTH_TYPE_LOGIN, AUTH_TYPE_PLAIN);
+            }
+            if (session.getConfiguration().saslConfiguration().isPresent()) {
```

Comment:
IMO `session.getConfiguration().enableOAuth()`  is better. 
Example: `org.apache.james.smtpserver.SMTPSaslTest#oauthWithNoTLSConnectShouldFail`